torch/csrc/jit/serialization/source_range_serialization.cpp

```diff
@@ -58,17 +86,35 @@ c10::IValue SourceRangeSerializer::serialize(const SourceRange& sr) {
       serialize_source(sr.source()), (int64_t)sr.start(), (int64_t)sr.end());
 }
 
+int64_t SourceRangeSerializer::store_text_and_get_index(
+    c10::string_view text_view) {
+  auto text_iter = text_to_idx_.find(text_view);
+  if (text_iter == text_to_idx_.end()) {
+    int64_t text_pos = static_cast<int64_t>(texts_.size());
+    texts_.emplace_back(text_view);
+    text_to_idx_[texts_.back().toStringView()] = text_pos;
+    return text_pos;
+  } else {
+    return text_iter->second;
+  }
+}
+
 c10::IValue SourceRangeSerializer::serialize_source(
     const std::shared_ptr<SourceView>& s) {
   if (serialized_sources.count(s)) {
     return serialized_sources.at(s);
   }
   c10::intrusive_ptr<c10::ivalue::Tuple> serialized;
   if (s == nullptr) {
-    serialized = c10::ivalue::Tuple::create({"", "", 0});
+    serialized = c10::ivalue::Tuple::create({-1, -1, 0});
   } else {
+    int64_t text_pos = store_text_and_get_index(s->text());
+    int64_t fname_pos = -1;
```

Comment:
I am a bit worried about using -1 as default value because Pickle doesn't have schema or verification anywhere. Could we set 0 as default value and push some string that says "not_available" in there so that a naive consumer of `fname_pos` wouldn't crash by looking up text table by idx without checking?